src/app/CommandHandler.h

```diff
@@ -80,6 +129,21 @@ class CommandHandler : public Command
 
     CHIP_ERROR AddClusterSpecificFailure(const ConcreteCommandPath & aCommandPath, ClusterStatus aClusterStatus) override;
 
+    /**
+     * IncRef will increase the inner refcount of the CommandHandler.
+     *
+     * Users should use CommandHandler::Handle for management the lifespan of the CommandHandler.
+     * DefRef should be released in reasonable time, and Close() should only be called when the refcount reached 0.
+     */
+    void IncRef();
+
+    /**
+     * DefRef is used by CommandHandler::Handle for decreasing the refcount of the CommandHandler.
+     * When refcount reached 0, CommandHandler will send the response to the peer and shutdown.
+     */
+    void DecRef();
+    size_t RefCount() const { return mRefCount; }
```

Comment:
These should all be protected, with Handle as a friend, no?

If we want to allow non-RAII refcounting, then these APIs need better names; it's not about holding a reference, but rather about saying that we have async work that needs to be waited on.  iirc @mrjerryjohns had better name suggestions for these.

Reply:
I think the names I had before were "IncrementHoldoff" and "DecrementHoldoff" (or something like that)